Index: class.php
===================================================================
--- class.php	(revisión: 83)
+++ class.php	(copia de trabajo)
@@ -40,7 +40,7 @@
 		}
 		else{
 			foreach($config as $key=>$val){
-				if(!isset($this->config[$key])) $this->config[$key] = $val;
+				if (!isset($this->config[$key])) $this->config[$key] = $val;
 			}
 			if ($this->config['secure'] == false) $this->Deny = false;
 			$password = explode(", ", $this->config['password']);
@@ -115,7 +115,7 @@
 		$this->bbcode = $this->config['bbcode'];
 	}
 	function isadmin(){
-		return isset($_COOKIE["secureid"]) && $_COOKIE["secureid"] == md5($this->config['admin']) ? true : $this->admin;
+		return (isset($_COOKIE['secureid']) && $_COOKIE['secureid'] == md5($this->config['admin']) ? true : $this->admin);
 	}
 	function getversion(){
 		$version = $this->cut_str($this->curl("https://code.google.com/p/vinaget-script/source/list", "", ""), '"detail?r=','"');
@@ -393,7 +393,7 @@
 			header("HTTP/1.1 404 Not Found");
 			$this->error1('erroracc');
 		}
-		if($job['proxy'] != 0 && $this->redirdl == true) {
+		if ($job['proxy'] != 0 && $this->redirdl == true) {
 			list($ip, ) = explode(":", $job['proxy']);
 			if($_SERVER['REMOTE_ADDR'] != $ip) { 
 				$this->wrong_proxy($job['proxy']);
@@ -462,13 +462,9 @@
 		}
 		while (strpos($header, "\r\n\r\n") === false);
 		/* debug */
-		if(isset($_GET['debug'])){
-			echo "<pre>";
-			echo "connected to : $hosts ".($job['proxy'] == 0 ? "" : "via {$job['proxy']}")."\r\n";
-			echo "$data\r\n\r\n";
-			echo "Server replied: \r\n";
-			echo $header;
-			echo "</pre>";
+		if ($this->isadmin() && isset($_GET['debug'])) {
+			// Uncomment next line for enable to admins this debug code.
+			// echo "<pre>connected to : $hosts ".($job['proxy'] == 0 ? '' : "via {$job['proxy']}")."\r\n$data\r\n\r\nServer replied: \r\n$header</pre>";
 			die();
 		}
 		/* debug */
@@ -1498,9 +1494,7 @@
 
 
 class Download {
-	
 	public $last = false;
-	
 	public function __construct ($lib, $site) {
 		$this->lib = $lib;
 		$this->site = $site;
Index: debug.php
===================================================================
--- debug.php	(revisión: 83)
+++ debug.php	(copia de trabajo)
@@ -1,18 +1,27 @@
 <?php
 ob_start();
-ob_implicit_flush (TRUE);
-ignore_user_abort (0);
-if( !ini_get('safe_mode') )set_time_limit(30);
+ob_implicit_flush(TRUE);
+ignore_user_abort(0);
+if (!ini_get('safe_mode')) set_time_limit(30);
 define('vinaget', 'yes');
 date_default_timezone_set('Asia/Jakarta');
-if(empty($_REQUEST['link'])){
-	echo "<center><br/><br/><br/><br/><br/><h2>DEBUG RESULT</h2></center>";
+
+require_once('class.php');
+$obj = new stream_get();
+
+// This debug code should be available only to admins.
+if ($obj->Deny || !$obj->isadmin()) {
+	setcookie('msg', 'debug.php :: Access Violation');
+	header('Location: index.php');
+	ob_end_flush();
+	exit;
+} elseif (!empty($_REQUEST['link'])) {
+	if (!empty($_REQUEST['proxy'])) $obj->proxy = $_REQUEST['proxy'];
+	echo '<pre>'.htmlspecialchars($obj->curl($_REQUEST['link'], (!empty($_REQUEST['cookie']) ? $_REQUEST['cookie'] : 0), (!empty($_REQUEST['post']) ? $_REQUEST['post'] : 0))) . '</pre>';
+} else {
+	echo '<center><br /><br /><br /><br /><br /><h2>DEBUG RESULT</h2></center>';
 }
-else{
-	include("class.php");
-	$obj = new stream_get();
-	if(!empty($_REQUEST['proxy'])) $obj->proxy = $_REQUEST['proxy'];
-	echo $obj->curl( $_REQUEST['link'], $_REQUEST['cookie'], $_REQUEST['post']);
-}
+
 ob_end_flush();
+
 ?>
\ No newline at end of file
Index: index.php
===================================================================
--- index.php	(revisión: 83)
+++ index.php	(copia de trabajo)
@@ -15,26 +15,26 @@
 */
 $using = isset($_COOKIE['using']) ? $_COOKIE['using'] : 'default';
 $using = isset($_REQUEST['using']) ? $_REQUEST['using'] : $using;
-setcookie("using", $using);
+setcookie('using', $using);
 ob_start();
-ob_implicit_flush (TRUE);
-ignore_user_abort (0);
-if( !ini_get('safe_mode') )set_time_limit(30);
+ob_implicit_flush(TRUE);
+ignore_user_abort(0);
+if (!ini_get('safe_mode')) set_time_limit(30);
 define('vinaget', 'yes');
-include("class.php");
+require_once('class.php');
 $obj = new stream_get(); 
 $obj->using = $using;
 $obj->current_version = 83;
 $obj->msg = false;
-if(isset($_COOKIE['msg']))$obj->msg = $_COOKIE['msg'];
-setcookie("msg", "");
+if (!empty($_COOKIE['msg'])) $obj->msg = htmlspecialchars($_COOKIE['msg']);
+setcookie('msg', '');
 $host = $obj->list_host;
 $skin = "skin/{$obj->skin}";
 error_reporting($obj->display_error ? E_ALL : 0);
 if ($obj->Deny == false){
-	include("{$skin}/function.php");
-	if(isset($_POST['urllist'])) $obj->main();
-	elseif(isset($_GET['infosv'])) showStat();
+	require_once("{$skin}/function.php");
+	if (isset($_POST['urllist'])) $obj->main();
+	elseif (isset($_GET['infosv'])) showStat();
 	elseif (!isset($_POST['urllist'])) include("{$skin}/index.php");
 }
 else include("{$skin}/login.php");
Index: proccess.php
===================================================================
--- proccess.php	(revisión: 83)
+++ proccess.php	(copia de trabajo)
@@ -1,81 +1,92 @@
 <?php
 ob_start();
-ob_implicit_flush (TRUE);
-ignore_user_abort (0);
-if( !ini_get('safe_mode') )set_time_limit(30);
+ob_implicit_flush(TRUE);
+ignore_user_abort(0);
+if (!ini_get('safe_mode')) set_time_limit(30);
 define('vinaget', 'yes');
 date_default_timezone_set('Asia/Jakarta');
 
-include("class.php");
-$obj = new stream_get(); 
+require_once('class.php');
+$obj = new stream_get();
 
-$page = isset($_GET['page']) ? $_GET['page'] : 'config';
+$page = !empty($_GET['page']) ? $_GET['page'] : '';
 
-if(!$obj->isadmin() && isset($_POST['config'])) { header("Location: index.php"); ob_end_flush(); exit; }
-elseif($page == 'config'){
-	if(isset($_POST['submit'])){
-		if(isset($_POST['config'])) {
-			foreach($_POST['config'] as $ckey => $cval){
-				if($cval == 'on' || $cval == 'off') $cval = $cval == 'on' ? true : false;
-				elseif(is_numeric($cval)) $cval = intval($cval);
+if ($obj->Deny || !$obj->isadmin() || empty($page)) {
+	setcookie('msg', 'proccess.php :: Access Violation');
+	header('Location: index.php');
+	ob_end_flush();
+	exit;
+}
+
+$msg = '';
+switch ($page) {
+	case 'config':
+		if (!empty($_POST['config']) && is_array($_POST['config'])) {
+			foreach ($_POST['config'] as $ckey => $cval) {
+				if ($cval == 'on' || $cval == 'off') $cval = $cval == 'on' ? true : false;
+				elseif (is_numeric($cval)) $cval = intval($cval);
 				else $cval = $cval;
 				$obj->config[$ckey] = $cval;
 			}
-			$msg = "Config Saved!";
+			$obj->save_json($obj->fileconfig, $obj->config);
+			$msg = 'Config Saved!';
 		}
-		else $msg = base64_encode("[".implode("][", $obj->config)."]");
-		$obj->save_json($obj->fileconfig, $obj->config);
-	}
-}
-elseif($page == 'cookie'){
-	if(isset($_POST['cookie'])){
-		$obj->save_cookies($_POST['type'], $_POST['cookie']);
-		$msg = "{$_POST['type']} Cookie Added!";
-	}
-	elseif(isset($_GET['del'])){
-		$obj->save_cookies($_GET['del'], "");
-		$msg = "{$_GET['del']} Cookie Deleted!";
-	}
-}
-elseif($page == 'account'){
-	if(isset($_POST['account'])){
-		if(empty($obj->acc[$_POST['type']])) {
-			$obj->acc[$_POST['type']]['max_size'] = $obj->max_size_default;
-			$obj->acc[$_POST['type']]['proxy'] = "";
-			$obj->acc[$_POST['type']]['direct'] = false;
+		break;
+
+	case 'cookie':
+		if (!empty($_POST['type']) && !empty($_POST['cookie'])) {
+			$obj->save_cookies($_POST['type'], $_POST['cookie']);
+			$msg = $_POST['type'] . 'Cookie Added!';
+		} elseif (!empty($_GET['del'])) {
+			$obj->save_cookies($_GET['del'], '');
+			$msg = $_GET['del'] . 'Cookie Deleted!';
 		}
-		$_POST['account'] = str_replace(" ", "", $_POST['account']);
-		$obj->save_account($_POST['type'], $_POST['account']);
-		$msg = "{$_POST['type']} Account Added!";
-	}
-	elseif(isset($_GET['del']) && isset($_GET['host'])){
-		$acc = $obj->acc[$_GET['host']]['accounts'];
-		unset($obj->acc[$_GET['host']]['accounts']);
-		foreach($acc as $key=>$val){
-			if($key == $_GET['del']) continue;
-			$obj->acc[$_GET['host']]['accounts'][] = $val;
+		break;
+
+	case 'account':
+		if (!empty($_POST['type']) && !empty($_POST['account'])) {
+			if (empty($obj->acc[$_POST['type']])) {
+				$obj->acc[$_POST['type']]['max_size'] = $obj->max_size_default;
+				$obj->acc[$_POST['type']]['proxy'] = '';
+				$obj->acc[$_POST['type']]['direct'] = false;
+			}
+			$_POST['account'] = str_replace(' ', '', $_POST['account']);
+			$obj->save_account($_POST['type'], $_POST['account']);
+			$msg = $_POST['type'] . 'Account Added!';
+		} elseif (!empty($_GET['del']) && !empty($_GET['host'])) {
+			$acc = $obj->acc[$_GET['host']]['accounts'];
+			unset($obj->acc[$_GET['host']]['accounts']);
+			foreach ($acc as $key => $val) {
+				if ($key == $_GET['del']) continue;
+				$obj->acc[$_GET['host']]['accounts'][] = $val;
+			}
+			$obj->save_json($obj->fileaccount, $obj->acc);
+			$msg = $_GET['host'] . 'Account Deleted!';
 		}
-		$obj->save_json($obj->fileaccount, $obj->acc);
-		$msg = "{$_GET['host']} Account Deleted!";
-	}
-}
-elseif($page == 'host'){
-	if(isset($_POST['host'])){
-		foreach($_POST['host'] as $key=>$val){
-			$obj->acc[$key]['max_size'] = $val['max_size'];
-			$obj->acc[$key]['proxy'] = $val['proxy'];
-			$obj->acc[$key]['direct'] = (isset($val['direct']) && $val['direct'] == "ON" ? true : false);
+		break;
+
+	case 'host':
+		if (!empty($_POST['host'])) {
+			foreach ($_POST['host'] as $key => $val) {
+				$obj->acc[$key]['max_size'] = $val['max_size'];
+				$obj->acc[$key]['proxy'] = !empty($val['proxy']) ? $val['proxy'] : '';
+				$obj->acc[$key]['direct'] = (isset($val['direct']) && $val['direct'] == 'ON' ? true : false);
+			}
+			ksort($obj->acc);
+			$obj->save_json($obj->fileaccount, $obj->acc);
+			$msg = 'Host Setting Changed!';
 		}
-		ksort($obj->acc);
-		$obj->save_json($obj->fileaccount, $obj->acc);
-		$msg = "Host Setting Changed!";
-	}
+		break;
+
+	default:
+		setcookie('msg', 'proccess.php :: Unknown Page Type');
+		header('Location: index.php');
+		ob_end_flush();
+		exit;
 }
-else{
-	header("Location: index.php");
-}
 
-setcookie("msg", empty($msg) ? "" : $msg);
-header("Location: index.php?id=admin&page={$page}");
+setcookie('msg', empty($msg) ? '' : $msg);
+header('Location: index.php?id=admin&page='.urlencode($page));
 ob_end_flush();
+
 ?>
\ No newline at end of file
